.github/workflows/build-and-release.yaml

```diff
@@ -0,0 +1,56 @@
+name: Build and Release
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: "21"
+          distribution: "adopt"
+
+      - name: Install system-wide LibSodium
+        run: sudo apt-get install -y libsodium-dev
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+
+      - name: Initialize submodule, generate jars, and build (including tests)
+        run: chmod +x initialize.sh && ./initialize.sh
+
+      - name: Get version from androidModule
```

Comment:
There are semantic-release plugins that could be useful here:

- Github Plugin(Github Release): https://github.com/semantic-release/github
- Gradle Plugin(Gradle Integration): https://www.npmjs.com/package/gradle-semantic-release-plugin